webapp/src/services/api.ts

```diff
@@ -195,25 +195,21 @@ export const api = createApi({
     updateDataActions: build.mutation<
       UtterancePatch[],
       {
-        persistentIds: UtterancePatch["persistentId"][];
-        newValue: DataAction;
-      } & GetUtterancesQueryState
+        utterancePatch: {
+          persistentId: UtterancePatch["persistentId"];
+          dataAction: DataAction;
+        }[];
```

Comment:
That seems redundant, instead:
```suggestion
        utterancePatch: UtterancePatch[];
```
Also, I think that should be called `body` directly:
```suggestion
        body: UtterancePatch[];
```

Reply:
I considered the above and simplified my recent commit as much as possible. But, I face some errors I am unsure of when I remove the async with"queryFn" and call the utterancePatch as body directly. But isn't the term "utterancePatch" more readable than calling body? Or I am unsure if there is any specific reason to call it that way.

Reply:
This is the "request body" as seen in the generated types.
```ts
    requestBody: {
      content: {
        "application/json": components["schemas"]["UtterancePatch"][];
      };
    };
```
I called it `body` in our helper `fetchApi` to align with the native `fetch()` API, which expects `body`.
I don't pretend it is a better name, just that it allows for that code simplification.
This mutation is similar to the config mutation that has a `body`: `updateConfig({ jobId, body: partialConfig })}`.

Reply:
Ah-ha! I think I know what issue you faced. The type `GetUtterancesQueryState` has a `body: never` field, so when renaming this to `body`, in combination with the other suggestion, you will have to change
```diff
-      } & GetUtterancesQueryState
+      } Omit<GetUtterancesQueryState, "body">
```

Reply:
Okay, Thank you for explaining to me why we use the term "body". It makes sense to my confusion now. 

Exactly, that was the error message I have been facing since last week. Thank you for the suggestion to solve the issue here.